packages/shared/src/mock/metadata.ts

```diff
@@ -209,5 +209,20 @@ export const mockMetadata = makeNxGroup(mockFilepath, 'NXroot', {
         }),
       }),
     ]),
+    makeNxGroup('nexus_root_no_default', 'NXroot', {
+      defaultPath: undefined,
+      children: [
+        makeNxGroup('nexus_entry_no_default', 'NXentry', {
+          defaultPath: undefined,
+          children: [
+            makeNxDataGroup('spectrum', {
+              signal: makeNxDataset('oneD', intType, [41]),
+              axes: { X: makeNxDataset('X', floatType, [41]) },
+              axesAttr: ['X'],
+            }),
+          ],
+        }),
+      ],
+    }),
```

Comment:
I don't think it makes much sense to have an `NXroot` group that is not actually at the root. Maybe just switch to `NXprocess`? Or even better, just remove the extra NX group to just test the `NXentry` without a default path -- I think that would still test the two behaviours we're interested in: 1) consider a group without a `default` attribute but with an `NX_class` attribute as an NX group; 2) look through the children to find the first NX group to use as default entity. The recursion part is not new, so there's no reason to test it here, I think.

Reply:
> I don't think it makes much sense to have an `NXroot` group that is not actually at the root.

Yes, this would not appear in a regular Nexus file but it was a way to test the *NXroot without default* usecase.

I can remove it if you want :smile:

Reply:
With my second implementation, testing `NXroot` specifically is indeed unnecessary. Might as well keep the mock data compliant, I reckon.